### PR TITLE
Closes #20, #31 feature request decision needed for initial envsetup.yml

### LINKED ISSUE #20: Feature Request:  Create directories from config
### Feature Idea

Take the `paths` specified in in the _envsetup.yml and create the necessary directories:

- If it's an absolute path, create in the specified destination
- If it's not in an absolute path, create the directories relative to a root_path parameter
- Default the root path to current working directory

### Relevant Input

something like `build_from_config(root_path = getwd())`

### Relevant Output

Print the directory tree of the created path? 

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by nicholas-masel**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page.
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging.
- [x] Address all merge conflicts and resolve appropriately.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by nicholas-masel** [comment]
@mstackhouse Snapshot test fails on R4.0, but passes on 4.1 and release.  4.1 and release have one fail but that is fixed in your PR.    I believe the checks are fine as is and can be merged if you agree w/the code/implementation.

File: R/init.R [hunk 1]
```diff
@@ -9,7 +9,42 @@
...
+    default_path <- system.file("default_envsetup.yml", package = "envsetup", mustWork = TRUE)
+
+    config_path <- file.path(project, "envsetup.yml")
+
+    file.copy(default_path, config_path)
+
+    usethis::ui_done(paste("Configuration file (envsetup.yml) has been written to", project))
+
+    # build out the default directory structure in the project
+    build_from_config(
```
Comment (nicholas-masel) [inline R/init.R:41]: Users should have the option to build the directories when they provide a custom configuration.  I'll adjust this code to provide this feature.

Comment (nicholas-masel) [reply R/init.R]: Added this featur with https://github.com/pharmaverse/envsetup/pull/33/commits/eaa4315425509bda6f4349a19bcfb5decc58b9f7

**Comment by nicholas-masel** [comment]
> @mstackhouse Snapshot test fails on R4.0, but passes on 4.1 and release. 4.1 and release have one fail but that is fixed in your PR. I believe the checks are fine as is and can be merged if you agree w/the code/implementation.

I found snapshot variants to cover this.
```r
r_version <- function() paste0("R", getRversion()[, 1:2])

test_that("can capture version nickname", {
  expect_snapshot(version$nickname, variant = r_version())
})
```
https://www.tidyverse.org/blog/2021/10/testthat-3-1/#snapshot-tests

The admiralci tests though use an old snapshot date for R4.0, where these variants are not available yet.  I've requested an update with admiral ci, https://github.com/pharmaverse/admiralci/issues/61.

I don't want to hold this PR back though due to this.

**Comment by mstackhouse** [comment]
> @mstackhouse Snapshot test fails on R4.0, but passes on 4.1 and release. 4.1 and release have one fail but that is fixed in your PR. I believe the checks are fine as is and can be merged if you agree w/the code/implementation.

That's fine with the snapshot tests and part of the intent - since they change over time. you don't want to have to rewrite the messages constantly to maintain the tests

File: R/init.R [hunk 2]
```diff
@@ -1,15 +1,82 @@
...
+      config <- config::get(file = config_path)
+
+      if (!exists("paths", where = config)) {
+        usethis::ui_oops("No paths are specified as part of your configuration.  Update your config file to add paths.")
+        return(invisible())
+      }
+
+      paths <- unlist(config$paths, use.names = FALSE)
+
+      missing_directories <- !sapply(paths, dir.exists)
```
Comment (mstackhouse) [inline R/init.R:39]: So if you look at the docs for `sapply()` they indicate that it shouldn't be used in package development. Even though this is a controlled instance, I'd recommend that we replace this with vapply:
```suggestion
      missing_directories <- !vapply(paths, dir.exists, TRUE)
```

**Review by mstackhouse — CHANGES_REQUESTED** [review]
One small comment and then I'm good.

**Pull request merged**